# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.cpp
# Commit: 32eeee79d628
# Full Hash: 32eeee79d628dade8109454c636d5f8aaff0c93e
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 04:41:59
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 7 - Use getter methods to access mBrowserParent and mBrowserBridgeChild. r=farre
#   
#   This prepares nsFrameLoader for replacing mBrowserParent and mBrowserBridgeChild
#   with a common interface by making special case code use a getter method instead
#   of direct access.
# ==============================================================================

diff -r 07678a2fa9e7 -r 32eeee79d628 dom/ipc/BrowserParent.cpp
--- a/dom/ipc/BrowserParent.cpp	Wed Apr 24 23:52:47 2019 -0500
+++ b/dom/ipc/BrowserParent.cpp	Thu Apr 25 12:36:16 2019 -0500
@@ -257,8 +257,7 @@
   if (!aFrameLoader) {
     return nullptr;
   }
-  PBrowserParent* remoteBrowser = aFrameLoader->GetRemoteBrowser();
-  return static_cast<BrowserParent*>(remoteBrowser);
+  return aFrameLoader->GetBrowserParent();
 }
 
 /*static*/